src/lecture/seven/array/list/CustomArrayList.java

```diff
@@ -0,0 +1,141 @@
+package lecture.seven.array.list;
+
+public class CustomArrayList<V> implements List<V> {
+    private static final int STARTING_CAPACITY = 10;
+    private static final int EMPTY_CAPACITY = 0;
+    private static final Object[] EMPTY_ARRAY = {};
```

Comment:
do you think we could use type T instead of object here?

Reply:
> do you think we could use type T instead of object here?

No, we couldn't. Because EMPTY_ARRAY is a constant, that's used in empty array situations. If we change object to a type, it would need removing of a static modifier ("non-static type variable cannot be referenced from a static context"), and that's kinda removes whole point of field being constant. 
For me personally, using field as and object array, and then casting it to needed type is the way to go.